### PR TITLE
Adopt sbt-typelevel

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by rossabaker** [comment]
Still depends on a snapshot.  Not ready for primetime, even if green.

File: build.sbt [hunk 1]
```diff
@@ -129,36 +122,16 @@ lazy val site = project
...
-    baseVersion := "1.1",
+    tlBaseVersion := "1.2",
     crossScalaVersions := Seq("2.12.15", Scala213, "3.0.2"),
-    scalaVersion := crossScalaVersions.value.filter(_.startsWith("2.")).last,
-    versionIntroduced := Map(
-      "3.0.0-RC1" -> "1.0.0",
-      "3.0.0-RC2" -> "1.0.1",
-      "3.0.0-RC3" -> "1.1.3"
-    ),
     homepage := Some(url("https://github.com/typelevel/case-insensitive")),
```
Comment (armanbilge) [inline build.sbt:133]: This is set automatically too btw.

File: build.sbt [hunk 2]
```diff
@@ -129,36 +122,16 @@ lazy val site = project
...
     startYear := Some(2020),
-    licenses := Seq("Apache-2.0" -> url("https://www.apache.org/licenses/LICENSE-2.0.html")),
-    scmInfo := Some(
-      ScmInfo(
-        url("https://github.com/typelevel/case-insensitive"),
-        "git@github.com:typelevel/case-insensitive.git")),
     githubWorkflowTargetTags ++= Seq("v*"),
     githubWorkflowPublishTargetBranches := Seq(RefPredicate.StartsWith(Ref.Tag("v"))),
```
Comment (armanbilge) [inline build.sbt:136]: As are these.

File: build.sbt (hunk 1 above)
Comment (rossabaker) [reply build.sbt:133]: True, but it should probably be https://typelevel.org/case-insensitive/.

**Comment by rossabaker** [comment]
We lost the Sonatype variable declarations.  Do we not need those?

File: build.sbt [hunk 3]
```diff
@@ -6,7 +6,11 @@ val catsV = "2.7.0"
 val scalacheckV = "1.15.4"
 val disciplineMunitV = "1.0.9"
 
-enablePlugins(SonatypeCiReleasePlugin)
+enablePlugins(TypelevelPlugin)
```
Comment (armanbilge) [inline build.sbt:9]:
```suggestion
enablePlugins(TypelevelCiReleasePlugin)
```

We could also make this an auto-plugin.

**Comment by rossabaker** [comment]
Yeah, that's confusing.  I thought sbt-typelevel included all the doodads.

File: build.sbt (hunk 3 above)
Comment (rossabaker) [reply build.sbt]: Yeah, that's confusing.  I thought sbt-typelevel included all the doodads.

**Comment by rossabaker** [comment]
Well, looks like it worked!

**Pull request merged**